src/dotnet/install.sh

```diff
@@ -4,489 +4,133 @@
-
-    # Set architecture variable to current user's architecture (x64 or ARM64).
-    architecture="$(get_architecture_name_for_target_os)"
-
-    # Set VERSION to empty string to ensure jq includes all .NET versions in reverse sort below 
-    if [ "${DOTNET_DOWNLOAD_VERSION}" = "latest" ]; then
-        DOTNET_DOWNLOAD_VERSION=""
-    fi
-
-    dotnet_patchless_version="$(echo "${DOTNET_DOWNLOAD_VERSION}" | cut -d "." --field=1,2)"
-
-    set +e
-    dotnet_channel_version="$(curl -s "${DOTNET_CDN_FEED_URI}/dotnet/release-metadata/releases-index.json" | jq -r --arg channel_version "${dotnet_patchless_version}" '[."releases-index"[]] | sort_by(."channel-version") | reverse | map( select(."channel-version" | startswith($channel_version))) | first | ."channel-version"')"
-    set -e
-
-    # Construct the releases URL using the official channel-version if one was found.  Otherwise make a best-effort using the user input.
-    if [ -n "${dotnet_channel_version}" ] && [ "${dotnet_channel_version}" != "null" ]; then
-        dotnet_releases_url="${DOTNET_CDN_FEED_URI}/dotnet/release-metadata/${dotnet_channel_version}/releases.json"
-    else
-        dotnet_releases_url="${DOTNET_CDN_FEED_URI}/dotnet/release-metadata/${dotnet_patchless_version}/releases.json"
-    fi
-
-    set +e
-    dotnet_releases_json="$(curl -s "${dotnet_releases_url}")"
-    set -e
-    
-    if [ -n "${dotnet_releases_json}" ] && [[ ! "${dotnet_releases_json}" = *"Error"* ]]; then
-        dotnet_latest_version="$(echo "${dotnet_releases_json}" | jq -r --arg sdk_or_runtime "${sdk_or_runtime}" '."latest-\($sdk_or_runtime)"')"
-        # If user-specified version has 2 or more dots, use it as is.  Otherwise use latest version.
-        if [ "$(echo "${DOTNET_DOWNLOAD_VERSION}" | grep -o "\." | wc -l)" -lt "2" ]; then
-            DOTNET_DOWNLOAD_VERSION="${dotnet_latest_version}"
-        fi
-
-        dotnet_download_details="$(echo "${dotnet_releases_json}" |  jq -r --arg sdk_or_runtime "${sdk_or_runtime}" --arg dotnet_version "${DOTNET_DOWNLOAD_VERSION}" --arg arch "${architecture}" '.releases[]."\($sdk_or_runtime)" | select(.version==$dotnet_version) | .files[] | select(.name=="dotnet-\($sdk_or_runtime)-linux-\($arch).tar.gz")')"
-        if [ -n "${dotnet_download_details}" ]; then
-            echo "Found .NET binary version ${DOTNET_DOWNLOAD_VERSION}"
-            DOTNET_DOWNLOAD_URL="$(echo "${dotnet_download_details}" | jq -r '.url')"
-            DOTNET_DOWNLOAD_HASH="$(echo "${dotnet_download_details}" | jq -r '.hash')"
-            DOTNET_DOWNLOAD_NAME="$(echo "${dotnet_download_details}" | jq -r '.name')"
-        else
-            err "Unable to find .NET binary for version ${DOTNET_DOWNLOAD_VERSION}"
-            exit 1
-        fi
+fetch_latest_sdk_version() {
+    local latest_STS=$(fetch_latest_sdk_version_in_channel "STS")
+    local latest_LTS=$(fetch_latest_sdk_version_in_channel "LTS")
+    if [[ "$latest_STS" > "$latest_LTS" ]]; then
```

Comment:
From https://github.com/devcontainers/features/pull/628#issuecomment-1646547165
> Running dotnet-install.sh without any arguments will always give you the latest LTS release. 

The default option of this Feature is `latest`, doesn't it mean it could either give LTS or STS depending on whichever is most recent? eg. The `latest` should provide `.NET 7` (which is STS) 


https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#cadence
![image](https://github.com/devcontainers/features/assets/24955913/a8a8884d-9c6f-4233-af6c-0a02d8666562)

Reply:
I think the code is correctly implemented ^, thanks!  👍